src/modules/accommodations/services/accommodations.service.ts

```diff
@@ -1,19 +1,43 @@
 import { Injectable } from '@nestjs/common';
 import { InjectRepository } from '@nestjs/typeorm';
 import { Accommodation } from '../entities/accommodations.entity';
+import { ListAccommodationsParamsDto } from '../dto/requests/list-accommodations-params.dto';
+import { Repository } from 'typeorm';
+import { SortOrder } from '#/shared/constants/sort-order.constant';
 
 @Injectable()
 export class AccommodationsService {
   constructor(
-    @InjectRepository(Accommodation) private accommodationRepo: Accommodation,
+    @InjectRepository(Accommodation)
+    private accommodationRepository: Repository<Accommodation>,
   ) {}
 
   create() {
     return 'This action adds a new Accommodation';
   }
 
-  findAll() {
-    return `This action returns all Accommodation`;
+  async listAccommodations(searchParams: ListAccommodationsParamsDto) {
+    const page = searchParams.page ?? 1;
```

Comment:
Create a util function to actually map the search params to DB query

Reply:
done